proj4/src/main/scala/geotrellis/proj4/CRS.scala

```diff
@@ -72,17 +60,14 @@ object CRS {
     * @return              The specified CoordinateReferenceSystem
     */
   def fromString(name: String, proj4Params: String): CRS =
-    new CRS {
-      val proj4jCrs: CoordinateReferenceSystem = crsFactory.createFromParameters(name, proj4Params)
-    }
+    new CRS { val proj4jCrs: CoordinateReferenceSystem = crsFactory.createFromParameters(name, proj4Params) }
```

Comment:
In proj4 1.1 or 2.0 (whatever), can we consider making these `case` classes instead of anonymous ones? Make for a bit more natural serialization, easier debugging, etc.

Reply:
@metasim it would be geotrellis change, sure I'll do it :+1:

Reply:
@metasim do you have any preferred idea / class hierarchy for the entire CRS story?

Reply:
@pomadchin I hadn't thought about it until you asked... how's this for a strawdog:

![image](https://user-images.githubusercontent.com/131013/63477484-4e824280-c453-11e9-9a2a-7f31139cce9b.png)

When `fromWKT` or `fromProj4String` or `getEpsgCode` you'd get an instance of one of the intermediate classes.

Reply:
@metasim can you bring this conversation into a separate issue?